### PR TITLE
Feature/multi config

### PR DESCRIPTION
**Pull request by skovbakke**
Added target for building multiple `.wic` files for an entire cluster. This is a temporary solution in order to combine our yocto setup with the configurations in the mukube-configurator.

* Integration of the mukube-configurator project. 
    * Down the line the mukube-configurator project should be replaced with configuration generated using the yocto build system.
* Creation of multiple wks-files.
    * Multiple wks-files are generated based on the config partitions output by the mukube-configurator recipe.
* Multiple invocations of `wic`
    * We run `wic` once for each wks file. The only difference between the generated wic files are the config partitions.

### CODE REVIEW

File: meta-k8s-setup/recipes-k8s-testing/k8s-testing/k8s-testing.bb [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
 
 DEPENDS = "e2fsprogs-native"
 
-do_install(){
-    rm -f ${T}/config.ext4
-    mkdir ${T}/config-partition/ -p
-    cd ${WORKDIR} && tar -cvf ${T}/config-partition/config.tar.gz InitConfiguration.yaml 
-    cd ${T} && mkfs.ext4 -d config-partition config.ext4 1G
+inherit deploy
+
```
Comment (jacobcrawford) [inline meta-k8s-setup/recipes-k8s-testing/k8s-testing/k8s-testing.bb:12]: Fix line indents in do_compile

**Review by jacobcrawford — CHANGES_REQUESTED** [review]
Nothing much. Looks good.

File: meta-k8s-cluster/recipes-mukube/images/mukube-cluster-image.bb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	# Code is adapted from the IMAGE_CMD_wic in image_types_wic.bbclass
+	cluster_wic="${WORKDIR}/cluster-wic"
+	tmp_wic="${WORKDIR}/tmp-wic"
+
+	# Iterates over the wks files generated by the do_create_wks task
+	for wks in ${WORKDIR}/cluster-wks/*
+	do
+		if [ -e "$tmp_wic" ]; then
+			# Ensure we don't have any junk leftover from a previously interrupted
+			# do_image_wic execution
```
[the remark this answers is not in the document]
Comment (skovbakke) [reply meta-k8s-cluster/recipes-mukube/images/mukube-cluster-image.bb:45]: It is present in the `IMAGE_CMD_wic` command, so i assume it is a good thing to do. I believe interrupt means `^C` to stop the execution.

**Comment by skovbakke** [comment]
Tests pass 
```
NOTE: Executing Tasks
RESULTS:
RESULTS - config.ConfigPartitionTest.test_config_present: PASSED (0.25s)
RESULTS - kubeadm.KubeadmInitTest.test_kubeadm_init: PASSED (71.94s)
SUMMARY:
mukube-test-image () - Ran 2 tests in 72.191s
mukube-test-image - OK - All required tests passed (successes=2, skipped=0, failures=0, errors=0)
NOTE: Tasks Summary: Attempted 310 tasks of which 308 didn't need to be rerun and all succeeded.
```

**Comment by klausenbusk** [comment]
Let's get https://github.com/distributed-technologies/mukube/pull/27 in first.

**Comment by skovbakke** [comment]
#27 is now merged.

**Review by jacobcrawford — APPROVED** [review]
lgtm

**Pull request merged**